pkg/nirmata/resource_cluster_registered.go

```diff
@@ -72,13 +116,137 @@ func resourceClusterRegistered() *schema.Resource {
+			return id, nil
+		}
+	}
+	return "", fmt.Errorf("[ERROR] - id not found for %s/%s", owner_type, owner_name)
+}
+
+func makeAccessControlListObj(ownerInfo []interface{}, users, teams []map[string]interface{}) (map[string]interface{}, error) {
+	var accessControlListObj map[string]interface{}
+	// Set owner_type to user as the default when no owner_info is provided
+	var owner_type string = "user"
+	var owner_name string
+
+	if ownerInfo != nil && len(ownerInfo) != 0 {
+		for _, o := range ownerInfo {
+			elem, ok := o.(map[string]interface{})
+			if ok {
+				owner_type = elem["owner_type"].(string)
+				owner_name = elem["owner_name"].(string)
+				owner_id, err := fetchOwnerId(owner_type, owner_name, users, teams)
+				if err != nil {
+					log.Printf("[ERROR] - No entity of type %s with name %s found in the cluster", owner_type, owner_name)
+					return nil, fmt.Errorf("[ERROR] - Invalid owner '%s/%s' provided", owner_type, owner_name)
+				}
+				accessControlListObj = map[string]interface{}{
+					"modelIndex": "AccessControlList",
+					"ownerType":  owner_type,
+					"ownerName":  owner_name,
+					"ownerId":    owner_id,
+				}
+			}
+		}
+	} else {
+		log.Printf("[DEBUG] - owner_info is empty")
+		accessControlListObj = map[string]interface{}{
+			"modelIndex": "AccessControlList",
+			"ownerType":  owner_type,
+		}
+	}
+	return accessControlListObj, nil
+}
+
+func makeAccessControlList(accessControlList []interface{}, users, teams []map[string]interface{}) ([]interface{}, error) {
+	var acObArr = make([]interface{}, 0)
+
+	if accessControlList != nil {
+		for _, ac := range accessControlList {
+			log.Printf("[DEBUG] - ac %v", ac)
+			elem, ok := ac.(map[string]interface{})
+			if ok {
+				entity_type := elem["entity_type"].(string)
+				entity_name := elem["name"].(string)
+				entity_id, err := fetchOwnerId(entity_type, entity_name, users, teams)
+				if err != nil {
+					log.Printf("[ERROR] - No entity of type %s with name %s found in the cluster", entity_type, entity_name)
+					return nil, fmt.Errorf("[ERROR] - Invalid entity '%s/%s' provided", entity_type, entity_name)
+				}
+				acOb := map[string]interface{}{
+					"modelIndex": "AccessControl",
+					"entityType": entity_type,
+					"entityName": entity_name,
+					"permission": elem["permission"],
+					"entityId":   entity_id,
+				}
+				log.Printf("[DEBUG] - Printing accessControlList - acOb %v", acOb)
+				acObArr = append(acObArr, acOb)
+			}
+		}
+	}
+	return acObArr, nil
+}
+
 func resourceClusterRegisteredCreate(d *schema.ResourceData, meta interface{}) error {
 	apiClient := meta.(client.Client)
 	name := d.Get("name").(string)
 	labels := d.Get("labels")
 	endpoint := d.Get("endpoint").(string)
 	clusterType := d.Get("cluster_type").(string)
 
+	users, err := getUsers(apiClient)
+	if err != nil {
+		log.Printf("[ERROR] - failed to retrieve users: %v", err)
+		return fmt.Errorf("[ERROR] - Fetching users failed")
```

Comment:
Consider leaving out [ERROR] in fmt.Errorf calls, since the place where the log.Println is done, would mostly add [ERROR].

Reply:
Resolved